src/components/ReportingPayments.js

```diff
@@ -1,11 +1,85 @@
 import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import _ from 'ramda';
+import $ from '../money';
+import moment from 'moment';
 
 class ReportingPayments extends Component {
+  constructor(props) {
+    super(props);
+    this.transactionsByDate = this.transactionsByDate.bind(this);
+    this.paymentsByDate = this.paymentsByDate.bind(this);
+    this.renderPaymentsByDate = this.renderPaymentsByDate.bind(this);
+    this.renderHeader = this.renderHeader.bind(this);
+  }
+
+  dateRange() {
+    const today = moment().startOf('day');
+    return _.map((days) => moment(today).subtract(days, 'days'), _.reverse(_.range(0, 7)));
+  }
+
```

Comment:
Getting the transactions / payments by date and type through these next 3 functions feels kinda sloppy.. wondering if it would be better to query db using date range on select?